Source/WebKit/UIProcess/API/wpe/PageClientImpl.cpp

```diff
@@ -233,6 +233,27 @@ void PageClientImpl::doneWithTouchEvent(const NativeWebTouchEvent& touchEvent, b
             event->modifiers &= ~wpe_input_pointer_modifier_button1;
             page.handleMouseEvent(NativeWebMouseEvent(event, page.deviceScaleFactor()));
         },
+        [&](TouchGestureController::ContextMenuEvent& contextMenuEvent)
+        {
+            auto* event = &contextMenuEvent.event;
+
+            // Mouse motion towards the point of the context click.
+            event->type = wpe_input_pointer_event_type_motion;
+            page.handleMouseEvent(NativeWebMouseEvent(event, page.deviceScaleFactor()));
```

Comment:
Note: if #9688 is merged before this, we should pass `WebMouseEventSyntheticClickType::OneFingerTap` here to prevent emitting additional pointer events. If this is merged before, we could do that in #9688 or in a separate PR.